proposal.rst

```diff
@@ -239,7 +242,37 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+Community Bonding Period ( May 6- May 27 )
+- In this period, I will be studing the xapian codebase in detail and discuss with my mentor regarding the doubts I have so that once the coding period start I can actually start coding and will not stay confused by no tunderstanding the codebase itself.
+
+Coding Period ( May 27- August 19 ) (12 weeks)
+Before starting of the coding period I will discuss with my mentor and divide my project into 3 parts since there are 3 weeks of coding time. The main algorithm will be divided into 3 sub parts and once approved by the mentor I will go through the coding period as follows:
+
+Week 1: Code part 1 of the algorithm
+
+Week 2: Test for errors and debugging
+
+Week 3: Approve from the mentor
+
+Week 4: Document the progress
+
+Week 5: First Evalutaion Week- will review the code and documentation and after discussing with mentor will submit the evalutaion.
+
+Week 6: Code part 2 of the algorithm
+
+Week 7: Test for errors, debug and discuss with mentor
+
+Week 8: Document the progress
+
+Week 9: Second Evalutaion Week- will review the code and documentation and after discussing with mentor will submit the evalutaion.
+
+Week 10: Code part 3 of the algorithm 
+
+Week 11: Error testing and debugging
+
+Week 12: Discuss with mentor and document the entire progress in detail.
+
+Final Evalutaion- After the completion of the project, I will look into it once more to find any bugs or errors, will also go through the documentation, and approve with my mentor before the submission of my project.
```

Comment:
This is all written in incredibly vague terms - almost nothing here is usefully measurable.

At each evaluation the mentor has to say if you're behind schedule, on schedule, or ahead of schedule, but since none of the tasks are clearly described it would be pretty much impossible to say.  For example, I can't tell if you've completed "Code part 1 of the algorithm" when you haven't defined anywhere what "part 1 of the algorithm" actually is.

You need to dig in to the project you're proposing and the areas of the code you'll be working on to get an understanding for what is actually involved in what you are proposing to do.  Without you having done that, it's hard for you to write a realistic proposal for the project.

You also appear to be only proposing to tackle the part of the project idea about using the docid ranges, and not the part about running some experiments to try to tune the estimates.  You're welcome to take a project idea and adjust it by adding parts and dropping others (though the project still needs to make sense as a whole so it's a good idea to discuss with us if you plan to do this).  By taking just half of a project I suspect this probably ends up being too little work to fill 3 months (it's hard to judge though, as you haven't submitted a patch, so I don't have any tangible evidence of your coding abilities).